microservices/sp_authorization_microservice.py

```diff
@@ -0,0 +1,567 @@
+
+        if is_allowed_registration:
+            registration_link = facility_attributes.get(self.__REGISTRATION_LINK) # noqa
+            if registration_link:
+                logger.debug(
+                    f"Redirecting user to custom registration link '"
+                    f"{registration_link}' configured for service ('"
+                    f"{sp_entity_id}').")
+                context.state[self.name] = data.to_dict()
+                return Redirect(registration_link)
+
+            try:
+                registration_data = self.__get_registration_data(user,
+                                                                 facility,
+                                                                 sp_entity_id,
+                                                                 facility_attributes)  # noqa
+                if registration_data:
+                    skip_notification = sp_entity_id in \
+                                        self.__skip_notification_sps
+                    self.register(context, data, registration_data,
+                                  skip_notification)
+                    return
+                logger.debug(f"No VO is available for registration into groups"
+                             f" of resources for service '{sp_entity_id}'")
+            except Exception as ex:
+                logger.warning(
+                    "Caught an exception, user will be redirected to "
+                    "unauthorized for security reasons")
+                logger.debug(ex)
+        else:
+            logger.debug(
+                f"User is not member of any assigned groups of resources for "
+                f"service ('{sp_entity_id}'). Registration to the groups is "
+                f"disabled.")
+
+        return self.unauthorized()
+
+    def __get_sp_attributes(self, facility: Facility) -> dict[str, Union[
+            str, Optional[int], bool, List[str], dict[str, str]]]:
+        """
+        Fetches attributes required for user registration into VOs from given
+        facility.
+
+        @param facility: Facility which attributes should be extracted
+        @return: attributes of given a facility necessary for registration
+        """
+        attr_names = [self.__check_group_membership_attr,
+                      self.__vo_short_names_attr]
+
+        if self.__ALLOW_REGISTRATION_ATTR:
+            attr_names.append(self.__ALLOW_REGISTRATION_ATTR)
+
+        if self.__REGISTRATION_LINK_ATTR:
+            attr_names.append(self.__REGISTRATION_LINK_ATTR)
+
+        result = {}
+
+        facility_attrs = self.__adapters_manager.get_facility_attributes(
+            facility, attr_names)
+
+        if self.__check_group_membership_attr in facility_attrs:
+            result[self.__CHECK_GROUP_MEMBERSHIP] = facility_attrs[
+                                                        self.__check_group_membership_attr] or False  # noqa
```

Comment:
```suggestion
        result[self.__CHECK_GROUP_MEMBERSHIP] = facility_attrs.get(self.__check_group_membership_attr, False)
```
shorter, applies to all attributes like this